lib/userAgent.js

```diff
@@ -34,6 +36,11 @@ module.exports = function (userAgentString) {
   this.uaOSVersion = getVersion(userAgentData.os) || null
   this.uaDeviceType = getDeviceType(userAgentData) || null
 
+  if (! this.uaBrowser && ! this.uaOS) {
+    // In the worst case, fall back to a truncated user agent string
+    this.uaBrowser = truncate(userAgentString || '', log)
+  }
+
```

Comment:
@philbooth since it looks like we'll be waiting O(week) for final word on the PII implications of this, what do you think about removing the user-agent-truncation stuff into a follow-up PR and landing the rest of this for train-65?  ISTM that it wouldn't make things any worse than they already are for badly-formed UA strings, and will recover whenever the truncation code is landed.